src/runners/PlaywrightTestRunner.ts

```diff
@@ -0,0 +1,112 @@
+import { join } from "path";
+import { debug, WorkspaceFolder } from "vscode";
+import { ITestRunnerInterface } from "../interfaces/ITestRunnerInterface";
+import { ConfigurationProvider } from "../providers/ConfigurationProvider";
+import { TerminalProvider } from "../providers/TerminalProvider";
+
+export class PlaywrightTestRunner implements ITestRunnerInterface {
+  public name: string = "playwright";
+  public path: string = join("node_modules", ".bin", this.name);
+  public terminalProvider: TerminalProvider = null;
+  public configurationProvider: ConfigurationProvider = null;
+
+  constructor(
+    configurationProvider: ConfigurationProvider,
+    terminalProvider: TerminalProvider,
+    path?: string
+  ) {
+    this.terminalProvider = terminalProvider;
+    this.configurationProvider = configurationProvider;
+
+    if (path) {
+      this.path = path;
+    }
+  }
+
+  public runTest(
+    rootPath: WorkspaceFolder,
+    fileName: string,
+    testName: string
+  ) {
+    const additionalArguments = this.configurationProvider.additionalArguments;
+    const environmentVariables =
+      this.configurationProvider.environmentVariables;
+
+    const command = `${
+      this.path
+    } test -g "${testName}" ${additionalArguments} ${this.transformFileName(
+      fileName
+    )}`;
+
+    const terminal = this.terminalProvider.get(
+      { env: environmentVariables },
+      rootPath
+    );
+
+    terminal.sendText(command, true);
+    terminal.show(true);
+  }
+
+  public debugTest(
+    rootPath: WorkspaceFolder,
+    fileName: string,
+    testName: string
+  ) {
+    const additionalArguments = this.configurationProvider.additionalArguments;
+    const environmentVariables =
+      this.configurationProvider.environmentVariables;
+    const skipFiles = this.configurationProvider.skipFiles;
+
+    /*debug.startDebugging(rootPath, {
+      preLaunchTasks: [
+        {
+          label: "Start Test",
+          type: "shell",
+          cwd: rootPath,
+          env: {
+              ...{PLAYWRIGHT_CHROMIUM_DEBUG_PORT: 9222, PWDEBUG: true},
+              ...environmentVariables
+          },
+          command: join(rootPath.uri.fsPath, this.path),
+          args: [
+            "test",
+            "-g",
+            testName,
+            ...additionalArguments.split(" "),
+            this.transformFileName(fileName),
+          ],
+
+        }
+      ],
+      name: "Debug Test",
+      request: "attach",
+      type: "pwa-chrome",
+      port: 9222
+    });*/
```

Comment:
If the code does not do anything useful, I think it should not be commited to the repo.